core/trino-main/src/test/java/io/trino/sql/query/TestUnnest.java

```diff
@@ -241,4 +241,21 @@ public void testNullRows()
                 + "     AS t(a) "
                 + "     CROSS JOIN UNNEST(a) t(x, y)");
     }
+
+    @Test
+    public void testCorrelatedUnnest()
```

Comment:
Does this fail without the fix? TestUnnest uses LocalQueryRunner, which doesn't use exchanges.

Reply:
Right. I moved the test to `AbstractTestEngineOnlyQueries`.